src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -124,24 +173,77 @@ export default class ReplyField extends React.Component<Props, State> {
         setCursorPosition(editorState.getSelection().getFocusOffset());
     };
 
+    stopDefaultEvent = (event: React.SyntheticEvent): void => {
+        event.preventDefault();
+        event.stopPropagation();
+    };
+
+    setPopupListActiveItem = (index: number): void => this.setState({ activeItemIndex: index });
+
+    handleReturn = (event: React.KeyboardEvent): DraftHandleValue => {
+        const { activeItemIndex, popupReference } = this.state;
+
+        if (!popupReference) {
+            return 'not-handled';
+        }
+
+        this.stopDefaultEvent(event);
+        this.handleSelect(activeItemIndex);
+
+        return 'handled';
+    };
+
+    handleArrow = (event: React.KeyboardEvent): void => {
+        const { activeItemIndex, editorState, popupReference } = this.state;
+        const collaborators = this.getCollaborators(getActiveMentionForEditorState(editorState));
+
+        if (!popupReference || !collaborators.length) {
+            return;
+        }
+
+        this.stopDefaultEvent(event);
+
+        let nextIndex = 0;
+
+        if (event.key === 'ArrowDown') {
```

Comment:
Can these be separate event handlers?

Reply:
> Can these be separate event handlers?

They were separate in my last commit but Conrad thought they were confusing. I also think they looked weird. So I combined the two arrows handlers. Is there a convention or something they have to be separate?

Reply:
https://github.com/box/box-annotations/pull/484/commits/eb20b7fe9093f52c07e18e08e088c1f9338f40fb#diff-ce0f2aa5a0bf66aefefef7061e8dfa40R214-R222

Reply:
The editor provides separate callback props, so having to check the key seems unnecessary. Maybe something like this, though the `handleArrow` return value is a little awkward:

```
    handleArrow = (event: React.KeyboardEvent): number => {
        const { popupReference } = this.state;
        const { length } = this.getCollaborators();

        if (!popupReference || !length) {
            return 0;
        }

        event.preventDefault();
        event.stopPropagation();

        return length;
    };

    handleDownArrow = (event: React.KeyboardEvent): void => {
        const { activeItemIndex } = this.state;
        const length = this.handleArrow(event);

        if (length) {
            this.setPopupListActiveItem(activeItemIndex === length - 1 ? 0 : activeItemIndex + 1);
        }
    };

    handleUpArrow = (event: React.KeyboardEvent): void => {
        const { activeItemIndex } = this.state;
        const length = this.handleArrow(event);

        if (length) {
            this.setPopupListActiveItem(activeItemIndex === length - 1 ? 0 : activeItemIndex - 1);
        }
    };
```